File: imu_calib_python/imu_calib/runtime/get_accel_bias_from_temperature.py
from __future__ import annotations

import numpy as np

from imu_calib.runtime.get_bias_from_temperature import get_bias_from_temperature


def get_accel_bias_from_temperature(
    temp: np.ndarray | None,
    ba_const: np.ndarray,
    temp_model: dict | None = None,
) -> tuple[np.ndarray, dict]:
    """Evaluate ba(T) with constant-bias fallback."""

    return get_bias_from_temperature(temp, ba_const, temp_model, target_name="ba")

File: imu_calib_matlab/src/runtime/get_accel_bias_from_temperature.m
function [baT, info] = get_accel_bias_from_temperature(temp, baConst, tempModel)
%GET_ACCEL_BIAS_FROM_TEMPERATURE Evaluate ba(T) with constant-bias fallback.

[baT, info] = get_bias_from_temperature(temp, baConst, tempModel, 'target_name', 'ba');
end
